UnitTests/ScientistTests.cs

```diff
@@ -121,5 +122,103 @@ public void AnExceptionReportsDuration()
             Assert.True(((InMemoryObservationPublisher)Scientist.ObservationPublisher).Observations.First(m => m.Name == "failure").ControlDuration.Ticks > 0);
             Assert.True(((InMemoryObservationPublisher)Scientist.ObservationPublisher).Observations.First(m => m.Name == "failure").CandidateDuration.Ticks > 0);
         }
+
+        [Fact]
+        public void RunsBothBranchesOfTheExperimentWithResultComparisonSetAndReportsSuccess()
+        {
+            bool candidateRan = false;
+            bool controlRan = false;
+
+            // We introduce side effects for testing. Don't do this in real life please.
+            // Do we do a deep comparison?
+            Func<ComplexResult> control = () => { controlRan = true; return new ComplexResult {Count = 10, Name = "Tester"}; };
```

Comment:
Now that we are using NSubstitute, want to take a crack at converting this test to use that? Check out the other tests for examples.